### DISCUSSION TITLE
Hosting these files?

### DISCUSSION THREAD

**Discussion by jonberkery-work** [CMSgov/price-transparency-guide#131] [General]
Has anyone determined how they will host\store the in-network, allowed-amount, and prescription drug files for the file consumer?

How many files are you expecting to host?

What are the typical file sizes for the three file you are estimating?

How are you planning to transfer the file to the file consumer?

**Comment by ghost** [comment]
@jonberkery-work Rough estimate barring a schema change to make an array of plan related root schema fields, would be
In-Network Rates could be between 2,500 - 3,000 files with an average file size of 100GB
Allowed Amounts are TBD for # files and size
Prescription Drugs could be 100s to 1000s and average file size size 120GB

We are providing URLs in pages from our site under a location from our front page in the same philosophy as done for the Interoperability mandate.

**Reply by bryanbriegel** [reply]
Incidentally, I modeled counts of files for a national issuer - just their QHPs - and under the current schema and file naming requirements - plan level - they would have over 1.7K sets of files, (which I thought was a lot until seeing your expected metrics.) If reported at the product level, 9 sets.

**Reply by ghost** [reply]
@bryanbriegel Yes, the In-Network Rates number would dramatically change at the product level. Our estimate could be as low as 12-32

**Reply by jonberkery-work** [reply]
@bryanbriegel - Do you have any estimate on the size of 1 set of the 1.7K sets of files?

For my issuer we will have roughly 2.7K sets of files.

Multiplying that out, that is 8.1K individual files\links (in-network, allowed amounts, pharmacy) ... for us.

**Reply by bryanbriegel** [reply]
Unfortunately I do not - just pulled their 2021 HIOS Plan IDs for their individual plans and modeled using Product (10byte) vs Plan level 14 byte.  (There is an aggregation clause in the rule around the allowed amount file, though it was not finalized exactly as proposed does allow combining multiple plans into an allowed amount file, though the current naming convention doesnt seem to support this approach)

**Comment by nhartmanNWA** [comment]
Hello,

Is a TPA required to host the MRF's on their company site for the various group health plans they administer or would it be compliant to create a "sub-site" by group health plan and host them there so that the TPA would not have to essentially list on their site all of their clients?  The sub-site would contain the group health plan name and reference to the MRF's. thus you could find it through an internet search, as well as links to the MRF's.

**Reply by Devyandu** [reply]
The Payer is responsible for complying with the regulation and providing a link to the files on their public website.  While the Payer needs to provide the link on their website, there is nothing in the regulation requiring that the physical files must be hosted on the same web host. 
They could be in cloud storage account, CDN or on the site of a vendor or partner that produced the files for them,

With your example, the TPA is the Reporting Entity. Even of the TPA hosts the file for several payers, the Payer still needs to have a public link on their website. (which could point to the files that the TPA hosts).

Here is a [somewhat similar discussion.](https://github.com/CMSgov/price-transparency-guide/discussions/257)

Here is an [FAQ](https://www.cms.gov/healthplan-price-transparency/resources/technical-clarification) that talks about reporting entities.

> **Question: The schema field titled, reporting_entity_name (for all machine-readable files) is defined as, "The legal name of the entity publishing the machine-readable file." Does this refer to the name of the entity publishing the machine-readable file on its website or the name of the entity that created the file (which could be included on the entity’s own website or on another entity’s website)?
Answer: The reporting_entity_name field identifies the entity that created the file, not the entity hosting the file (which would most likely be the plan/issuer). Here are some examples below:** 

1. An insurance carrier, ABC Health, creates an in-network rate file and publishes it on the public website, ABCHealth.com. In this case, the reporting_entity_name should contain "ABC Health".
2. ABC Health has its claims processed by an external vendor, Claims R Us, and the vendor creates the in-network rate file for ABC Health to publish on its website, ABCHealth.com. The reporting_entity_name should contain "Claims R Us".
3. ABC Health is providing a self-funded group, ACME Manufacturing, an in-network rate file for the self-funded group to publish on its website, ACMEManufact.com. The reporting_entity_name should contain "ABC Health". 

PS : I am not with CMS.
